### PR TITLE
Add error case testing and empty bytestring tests

### LINKED ISSUE #5: Add error hypothesis strategies.
New hypothesis tests that generate random data and try to decompress it.  Both our implementation and the `python-snappy` implementation should always either decompress to the same value or both throw an exception.

### LINKED ISSUE #7: Test empty byte string for both compress and decompress
explicit tests for feeding `b''` into `compress` and `decompress`.

### PR DESCRIPTION
**Pull request by pipermerriam**
fixes #5 
fixes #7

## What was wrong?

We also want to be sure that `py-snappy` always throws errors on values that `python-snappy` would error out on.

## How was it fixed?

Added hypothesis tests which try to decode random values and ensure that both implementations behave the same.

#### Cute Animal Picture

squirrel-pumpkin-06

### CODE REVIEW

**Comment by carver** [comment]
:grimacing: commits titled `d` don't meet our style guideline I think :rofl:

File: tests/core/test_libsnappy_compat.py [hunk 1]
```diff
@@ -20,3 +33,33 @@ def test_libsnappy_decompress_local_compressed(value):
...
+
+    try:
+        lib_result = libsnappy_decompress(value)
+    except LIB_SNAPPY_ERRORS:
+        lib_snappy_error = True
+    else:
+        lib_snappy_error = False
+
+    if py_snappy_error and lib_snappy_error:
+        pass
```
Comment (carver) [inline tests/core/test_libsnappy_compat.py]: We want to catch if one fails and the other doesn't, right? Something like:

```py
assert py_snappy_error is lib_snappy_error
```

File: tests/core/test_libsnappy_compat.py [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
+    CompressedLengthError = None
+
 
 MEGABYTE = 1000000
 
 
+#
+# Round trip value -> compress() -> decompress()
+#
 @given(value=st.binary(min_size=1, max_size=2 * MEGABYTE))
```
Comment (carver) [inline tests/core/test_libsnappy_compat.py]: Maybe `min_size` can go down to 0 now that we know it's supposed to work.

File: tests/core/test_round_trip.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 
 @given(value=st.binary(min_size=1, max_size=2 * MEGABYTE))
-@settings(max_examples=1000)
+@settings(max_examples=10000)
```
Comment (carver) [inline tests/core/test_round_trip.py]: Nice

File: tests/core/test_libsnappy_compat.py (hunk 1 above)
Comment (pipermerriam) [reply tests/core/test_libsnappy_compat.py]: There isn't parity between the errors they two libraries raise, only that they either always both do or do not raise an exception.  In the case where they both raise (the case being checked here), this is the only assertion I think we can make.

Comment (carver) [reply tests/core/test_libsnappy_compat.py]: These are both just `bool`s, a direct comparison should work, no? As is, I don't see how the test catches if one library fails and the other doesn't.

Comment (pipermerriam) [reply tests/core/test_libsnappy_compat.py]: You're right, I'll update
